### DISCUSSION TITLE
What about cookie consent?

### DISCUSSION THREAD

**Discussion by tikita** [datenstrom/community#599] [Ask a question] [answered]
Is there an extension for "cookie consent"?

Thanks, Tikita

**Comment by schulle4u** [comment, marked as answer]
Fortunately no. 😀 By default Yellow only creates authentication cookies for the edit extension, normal visitors should not be affected. However, some extensions might set 3rd-party cookies (Twitter, SoundCloud etc), I'm not sure if this could cause legal problems regarding user consent. Feel free to add more information, maybe we can find a solution.

**Reply by HelloLudger** [reply]
Happy to find this. Yes, those extensions would be a problem. Maybe we could get them to use the "2 clicks for privacy" principle: http://www.h-online.com/features/Two-clicks-for-more-privacy-1783256.html

That should include the appropriate warning for cookies.

**Reply by schulle4u** [reply]
Regarding link sharing / like buttons I suggest to have a look at the [Minishare extension](https://github.com/GiovanniSalmeri/yellow-minishare) by @GiovanniSalmeri. Its probably not as fancy as the Heise solution, but can be used completely without javascript. No data is being transfered to 3rd-party services until the user clicks a link. Actually my first extension for yellow was an implementation of the Shariff buttons by Heise, but I've abandoned it because of the huge JS/CSS overhead.

**Comment by markseuffert** [comment]
I like cookie-less websites. 

A website doesn’t have to set any cookies at all. From a technical point of view, it is not necessary to give a cookie to every user who visits your website. Cookies are only needed for certain actions, for example when someone creates a user account. Traditionally cookies have been used in three main areas: Tracking users, user management, session management. Ditch invasive services like Google Analytics and Facebook, avoid collecting personal data from visitors, then you won't need a cookie consent on every page.

https://www.scottcole.net/run-my-website-without-using-cookies-part1/
https://www.scottcole.net/run-my-website-without-cookies-part2/
https://creativerly.com/google-analytics-alternatives/
https://ethical.net/ethical/google-alternatives/

If this isn't your cup of coffee and you absolutely need cookies, have a look at #680.

**Reply by GiovanniSalmeri** [reply]
I fully agree with you (this is the reason why I added support for Matomo https://github.com/GiovanniSalmeri/yellow-analytics/commit/892f932129894dd0345cd4a7e51f121dd6b1c91b and put it in first position, and also the reason why I chose to use for https://github.com/GiovanniSalmeri/yellow-analytics a script which presents in visually identical prominence «Accept» and «Reject»). On the other side, I understand that there *are* legimate uses for tracking users (and now fortunately GDPR forces to declare clearly this): as in physical shops the owners observe their customers (how old are they? what do they look at? etc.), it is understandable that in e-commerce sites the owners want to know something about those who visit them: this is business. Or studying analytics (e.g. after the restructuring of a site) can greatly help to improve the usability. Etc. But those who add intrusive analytics without good reasons to their sites should be warned that they only annoy their visitors (and slow down the browsing, of course).
